### PR TITLE
Added hash mode 18500 sha1(md5(md5($pass)))

### LINKED ISSUE #1652: REQUEST New algo sha1(md5(md5($password)))
Hi,
	
Can you add new algo sha1(md5(md5($password))) for my own specific purpose?

Or maybe you can help me and say why  my code not work https://pastebin.com/GYUpvbBG

### PR DESCRIPTION
**Pull request by s3inlc**
Added hash mode 18500 supporting the sha1(md5(md5($pass))) algorithm, including tests for this hash mode.

closes #1652

### CODE REVIEW

**Comment by jsteube** [comment]
Thanks s3inlc!

There's a few things missing:

- Entry in docs/readme.txt
- Entry in extra/tab_completion/hashcat.sh
- Entry in src/usage.c

If you want you can also try the optimized kernels. Especially for fast hashes like this one it has a serious impact.

**Comment by s3inlc** [comment]
The entry in the readme should be done with the second commit. And the tab_completion is contained in the first commit. Did I miss something there?

I can give the optimized ones a try maybe at a later point.

**Comment by jsteube** [comment]
Many thanks!

File: src/interface.c [hunk 1]
```diff
@@ -28602,6 +28605,22 @@ int hashconfig_init (hashcat_ctx_t *hashcat_ctx)
                  hashconfig->dgst_pos2      = 2;
                  hashconfig->dgst_pos3      = 3;
                  hashconfig->st_hash        = ST_HASH_18400;
+                 hashconfig->st_pass        = ST_HASH_18500;
```
Comment (Naufragous) [inline src/interface.c:28608]: @s3inlc, ouch! :laughing:
GH-1821

Comment (s3inlc) [reply src/interface.c:28608]: uhh, indeed a very stupid mistake 🙈 
But I'm surprised that the tests didn't fail when I ran it.

Comment (Naufragous) [reply src/interface.c:28608]: Well, the tests for 18500 run just fine. It was the 18400 case branch that took the hit :sweat_smile:

**Pull request merged**